pypsa/io.py

```diff
@@ -901,6 +922,10 @@ def import_components_from_dataframe(network, dataframe, cls_name):
         logger.error("Error, new components for {} are not unique".format(cls_name))
         return
 
+    if cls_name in network.geo_components:
+        new_df["geometry"] = new_df["geometry"].apply(clean_geometry)
+        new_df = gpd.GeoDataFrame(new_df, geometry="geometry", crs=network.srid)
```

Comment:
just out of curiousity. What is `network.srid` ??

Reply:
![Uploading image.png…]()

Reply:
![image](https://github.com/PyPSA/PyPSA/assets/93286254/ad394f5f-4d57-4670-9aea-a0782aaada4e)